src/screens/UserProfile/Components/UserProfileGallery.tsx

```diff
@@ -1,97 +1,117 @@
 import { photo, playBtn, video } from '../../../svg/svg-xml-list';
 import { useStyles } from '../styles';
 import ImageView from '../../../components/react-native-image-viewing/dist';
+import { FeedRefType } from '../../CommunityHome';
 
 interface IUserProfileGallery {
   userId: string;
 }
 
-const UserProfileGallery: FC<IUserProfileGallery> = ({ userId }) => {
-  const { images, videos, getNextPage } = useGallery(userId);
-  const [showFullImage, setShowFullImage] = useState(false);
-  const [imageIndex, setImageIndex] = useState(0);
-  const [tabName, setTabName] = useState<TabName>(TabName.Photos);
-  const styles = useStyles();
-  const data = tabName === TabName.Photos ? images : videos;
+const UserProfileGallery = forwardRef(
+  ({ userId }: IUserProfileGallery, ref: React.Ref<FeedRefType>) => {
+    const [showFullImage, setShowFullImage] = useState(false);
+    const [imageIndex, setImageIndex] = useState(0);
+    const [tabName, setTabName] = useState<TabName>(TabName.Photos);
+    const dataType = tabName === TabName.Photos ? 'image' : 'video';
+    const { mediaFiles, getNextPage } = useGallery({
+      targetId: userId,
+      targetType: 'user',
+      dataType: dataType,
+      limit: 21,
+    });
+    const styles = useStyles();
+    const handleLoadMore = useCallback(() => {
+      getNextPage && getNextPage();
+    }, [getNextPage]);
+    useImperativeHandle(ref, () => ({
+      handleLoadMore,
+    }));
 
-  const onPressThumbnail = useCallback((index) => {
-    setImageIndex(index);
-    setShowFullImage(true);
-  }, []);
+    const onPressThumbnail = useCallback((index) => {
+      setImageIndex(index);
+      setShowFullImage(true);
+    }, []);
 
-  const renderGalleryContent = useCallback(
-    ({ item, index }) => {
-      return (
-        <TouchableOpacity onPress={() => onPressThumbnail(index)}>
-          <Image source={{ uri: item.uri }} style={styles.thumbnail} />
-          {item.dataType === 'video' && (
-            <View style={styles.playButton}>
-              <SvgXml xml={playBtn} width="50" height="50" />
-            </View>
-          )}
-        </TouchableOpacity>
-      );
-    },
-    [onPressThumbnail, styles.playButton, styles.thumbnail]
-  );
+    const renderGalleryContent = useCallback(
+      ({ item, index }) => {
+        return (
+          <TouchableOpacity onPress={() => onPressThumbnail(index)}>
+            <Image source={{ uri: item.uri }} style={styles.thumbnail} />
+            {item.dataType === 'video' && (
+              <View style={styles.playButton}>
+                <SvgXml xml={playBtn} width="50" height="50" />
+              </View>
+            )}
+          </TouchableOpacity>
+        );
+      },
+      [onPressThumbnail, styles.playButton, styles.thumbnail]
+    );
 
-  const renderGallery = useCallback(() => {
-    if (!data || data?.length === 0) {
-      const icon = tabName === TabName.Photos ? photo() : video();
+    const renderGallery = useCallback(() => {
```

Comment:
imo, we don't need `useCallback` here it's ok to use a condition rendering, wdyt?

Reply:
Sure, removed!